### PR TITLE
48 filter event element matrix

### LINKED ISSUE #48: Filter Event-Element matrix
The settings and some general heuristics (e.g. no keylog on merge button) should be used to filter the matrix, so that only useful and user-wanted event handlers exist on the page.

### PR DESCRIPTION
**Pull request by Exclaminator**
Will close #48 

Proof-of-concept for disabling specific behaviour of logging. Next week we will decide what the types of disabling that are useful are and I will implement them in this class.

### CODE REVIEW

File: src/octopeer-github/main/Settings/DoNotWatchSettings.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        }
+
+        return doNotWatchEvents;
+    }
+    /**
+     * Gets Elements not to Log, from the chrome storage.
+     * @returns {ElementSelectionBehaviourCreatable[]}
+     */
+    public getCombinations() {
+        let doNotWatchCombination: ElementXEventCreatable[];
```
Comment (mdingjan) [inline src/octopeer-github/main/Settings/DoNotWatchSettings.ts]: Isn't this local variable 'doNotWatchCombination' redundant?
Or will you modify this later on?

File: src/octopeer-github/main/Settings/DoNotWatchSettings.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+/**
+ * Class for indicating all internal settings of the application.
+ */
+const DoNotWatchSettings = new (class DoNotWatchSettings {
+    /**
+     * Gets Elements not to Log, from the chrome storage.
+     * @returns {ElementSelectionBehaviourCreatable[]}
+     */
+    public getElements() {
+        let doNotWatchElements: ElementSelectionBehaviourCreatable[];
```
Comment (mdingjan) [inline src/octopeer-github/main/Settings/DoNotWatchSettings.ts]: My same comment as below applies here.

File: src/octopeer-github/main/Settings/DoNotWatchSettings.ts (hunk 1 above)
Comment (Exclaminator) [inline src/octopeer-github/main/Settings/DoNotWatchSettings.ts]: Specific settings that remove certain combinations will be added next sprint.

File: src/octopeer-github/main/Settings/DoNotWatchSettings.ts (hunk 2 above)
Comment (Exclaminator) [inline src/octopeer-github/main/Settings/DoNotWatchSettings.ts]: My same comment as below applies here.

**Comment by mpsijm** [comment]
The proof of concept is correct, this is exactly what we need! Can be merged :)

**Comment by mdingjan** [comment]
I'm fine with the changes. :)
There will be some conflicts when merging this with the Options issue, but I'll resolve those. 

Ready for merging.

**Comment by MathiasMeuleman** [comment]
Small PR's are easy the review :D

Looks good, will merge!

**Pull request merged**